src/codecs/GroupChatMemberAdded.ts

```diff
@@ -0,0 +1,39 @@
+import { ContentCodec, ContentTypeId, EncodedContent } from '../index'
+
+export const ContentTypeGroupChatMemberAdded: ContentTypeId = {
+  typeId: 'groupChatMemberAdded',
+  authorityId: 'xmtp.org',
+  versionMajor: 1,
+  versionMinor: 0,
+  sameAs(id) {
+    return (
+      this.typeId === id.typeId &&
+      this.authorityId === id.authorityId &&
+      this.versionMajor === id.versionMajor &&
+      this.versionMinor === id.versionMinor
+    )
+  },
+}
+
+export type GroupChatMemberAdded = {
+  member: string
+}
+
+export class GroupChatMemberAddedCodec
+  implements ContentCodec<GroupChatMemberAdded>
+{
+  contentType = ContentTypeGroupChatMemberAdded
+
+  encode(content: GroupChatMemberAdded): EncodedContent {
+    return {
+      type: ContentTypeGroupChatMemberAdded,
+      parameters: {},
+      content: new TextEncoder().encode(JSON.stringify(content)),
+    }
+  }
+
+  decode(encodedContent: EncodedContent): GroupChatMemberAdded {
+    const json = new TextDecoder().decode(encodedContent.content)
+    return JSON.parse(json)
```

Comment:
We should at least do some runtime validation that the message actually looks anything like `GroupChatMemberAdded`. People can put anything in this JSON payload, and if we throw an error here it will be contained and the message will be ignored. If we wait until it gets accessed, it'll blow up later when it might just crash the whole app.

In a perfect world, these would all be protos.

Reply:
Someone doing something like:
```
conversation.send({member: {derp: true}} as any, { contentType: ContentTypeGroupChatMemberAdded })
```
Would probably break everyone in the group's app.